tests/test_feature_aggregate.py

```diff
@@ -299,6 +312,37 @@ def test_aggregate_region_with_components(simple_df):
     assert _df.check_aggregate_region(v, components=["foo"]) is None
 
 
+def test_agg_weight():
+    data = NEG_WEIGHTS_DF
+    variable = "Price|Carbon"
+    weight = "Emissions|CO2"
+    # negative weights should be dropped on default
+    test_1 = IamDataFrame(data).aggregate_region(
+        variable, weight=weight)
+    exp_1 = IamDataFrame(
+        pd.DataFrame([[np.nan, 5.25]], columns=[2005, 2010]),
+        model="model_a",
+        scenario="scen_a",
+        variable=variable,
+        unit="USD/tCO2",
+        region="World",
+    )
+    assert_iamframe_equal(test_1, exp_1)
```

Comment:
I know this is redundant but I get "ValueError: duplicate rows in 'data'" otherwise..